### PR TITLE
Add DragonWhale

### PR DESCRIPTION
**Pull request by Spandamn**

### CODE REVIEW

**Comment by ghost** [comment]
So idk how to tell u as DragonWhale never updated the OP. We're using gen 8 disguise instead of Alola's Disguise. Im sorry D:

**Comment by Spandamn** [comment]
rip
ill update in a bit

File: data/mods/ssb/moves.ts [hunk 1]
```diff
@@ -466,6 +466,39 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		shortDesc: "If Disguise intact or Mimikyu is the last remaining Pokemon, raises attack by 2; otherwise repairs Disguise and switches out.",
+		name: "Cloak Dance",
+		pp: 5,
+		priority: 0,
+		flags: {dance: 1},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, ['Teleport', 'Swords Dance'][this.random(2)], target);
```
Comment (ghost) [inline data/mods/ssb/moves.ts:484]: Uh, i dont think thats what dragonwhale intended as the animation. I think its dependant on whether disguise breaks or not.

File: data/mods/ssb/moves.ts [hunk 2]
```diff
@@ -466,6 +466,39 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		},
+		onHit(target, source) {
+			const flag = target.side.filter((ally: Pokemon) => ally === target || !ally.fainted).length > 1;
+			if (!target.volatiles['alolasdisguise'].busted || !flag) {
+				this.boost({atk: 2}, target);
+			} else {
+				this.add('-activate', target, 'move: Cloak Dance');
+				delete target.volatiles['alolasdisguise'].busted;
+				target.formeChange("Mimikyu");
```
Comment (ghost) [inline data/mods/ssb/moves.ts]:
```suggestion
				target.formeChange('Mimikyu');
```

File: data/mods/ssb/moves.ts [hunk 3]
```diff
@@ -466,6 +466,39 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+			}
+		},
+		selfSwitch: true,
```
Comment (ghost) [inline data/mods/ssb/moves.ts:496]: You need to disable this or else you are going to lose your swords dance boost!

File: data/mods/ssb/random-teams.ts [hunk 4]
```diff
@@ -112,6 +112,12 @@ export const ssbSets: SSBSets = {
 		signatureMove: 'Archangel\'s Requiem',
 		evs: {hp: 252, def: 128, spd: 128}, nature: 'Bold',
 	},
+	DragonWhale: {
+		species: 'Mimikyu', ability: 'Alola\'s Disguise', item: 'Life Orb', gender: 'M',
+		moves: ['Play Rough', 'Spectral Thief', 'Shadow Sneak'],
+		signatureMove: 'CloakDance',
```
Comment (ghost) [inline data/mods/ssb/random-teams.ts]:
```suggestion
		signatureMove: 'Cloak Dance',
```

**Pull request closed without merging**